protocol/messenger_communities.go

```
@@ -5173,9 +5173,9 @@ func (m *Messenger) startRequestMissingCommunityChannelsHRKeysLoop() {
 
 // getCommunityStorenode returns the active mailserver if a communityID is present then it'll return the mailserver
 // for that community if it has a mailserver setup otherwise it'll return the global mailserver
-func (m *Messenger) getCommunityStorenode(communityID ...string) peer.ID {
+func (m *Messenger) getCommunityStorenode(communityID ...string) peer.AddrInfo {
 	if m.transport.WakuVersion() != 2 {
-		return ""
+		return peer.AddrInfo{}
```

Maybe better to return an error explaining what happened?

---

Hmm, actually we can modify the code to not rely on the waku version, since @osmaczko removed wakuv1:

- `WakuVersion() != 2`: remove the entire condition
- `WakuVersion() == 1`: remove the entire condition
- `WakuVersion() == 2`: no need to check for the condition since it will always be `true`

---

> Hmm, actually we can modify the code everywhere to not rely on the waku version, since @osmaczko removed wakuv1:
> 
> * `WakuVersion() != 2`: remove the entire condition
> * `WakuVersion() == 1`: remove the entire condition
> * `WakuVersion() == 2`: no need to check for the condition since it will always be `true`

Sounds good! I would probably do it in a separate PR, but I agree that the condition is irrelevant now